services/grpc/internal/transformations/graph_objects.go

```diff
@@ -0,0 +1,36 @@
+package transformations
+
+type Node struct {
+	NodeId string `db:"nodeId"`
+	Doc string `db:"doc"`
+
+	EdgesOut string[] `db:"edgesOut"` // IDs of edges whose destination is this node
+	EdgesIn string[] `db:"edgesIn"`   // ID of edges whose source is this node
+
+	// Metadata
+	Date string `db:"date"`
+	Author string `db:"author"`
+}
+
+type Edge struct {
+	EdgeId string `db:"edgeId"`
+	SourceDoc string `db:"sourceDoc"`
+	TargetDoc string `db:"targetDoc`
+	Args FxnArgs `db:"args"` 				// f(SourceDoc, Args) -> TargetDoc
+	InverseArgs FxnArgs `db:"inverseArgs"`  // f(TargetDoc, InverseArgs) -> SourceDoc
+
+	SourceNodeId string `db:"sourceNodeId"`
+	TargetNodeId string `db:"targetNodeId"`
+
+	// Metadata
+	Date string `db:"date"`
+	Author string `db:"author"`
+}
+
+type Scope struct {
+	ScopeId string `db:"scopeId"`
+	
+	// Scope consists of a set of nodes, and thex set of edges between them
+    Nodes string[] `db:"nodes"` // Node IDs
+    Edges string[] `db:"edges"` // Edge IDs
```

Comment:
In theory would a Scope only need to include Edges if we want to ability to traverse the graph? I understand the role of Edges, less so Nodes.

Reply:
In theory you can choose just edges or just nodes and let the other set be defined by the one you chose. I included both because I expect it will be useful data for methods involving the scope, I can imagine use cases where we want to iterate over all nodes, etc

Reply:
Sure, works for me.

Reply:
TODO: Remove Edges from Scope. Keep a comment describing why for more context.